knimin/lib/tests/test_data_access.py

```diff
@@ -1171,6 +1171,137 @@ def test_delete_dna_plate(self):
+                'target_gene_region': '16S V4'},
+               {'id': 2, 'name': 'Primer plate 2', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 3, 'name': 'Primer plate 3', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 4, 'name': 'Primer plate 4', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 5, 'name': 'Primer plate 5', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 6, 'name': 'Primer plate 6', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 7, 'name': 'Primer plate 7', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 8, 'name': 'Primer plate 8', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'}]
+        self.assertEqual(db.get_targeted_primer_plates(), exp)
+
+    def test_prepare_targeted_libraries(self):
+        # Create a study
+        db.create_study(9999, title='LabAdmin test project', alias='LTP',
+                        jira_id='KL9999')
+        self._clean_up_funcs.append(partial(db.delete_study, 9999))
+
+        # Create some sample plates
+        pt = db.get_plate_types()[0]
+        plate_id = db.create_sample_plate('Test plate', pt['id'], 'test',
+                                          [9999])
+        self._clean_up_funcs.insert(
+            0, partial(db.delete_sample_plate, plate_id))
+        plate_id_2 = db.create_sample_plate('Test plate 2', pt['id'], 'test',
+                                            [9999])
+        self._clean_up_funcs.insert(
+            0, partial(db.delete_sample_plate, plate_id_2))
+
+        # Create DNA plates
+        robot = db.get_property_options("extraction_robot")[0]
+        kit = db.get_property_options("extraction_kit_lot")[0]
+        tool = db.get_property_options("extraction_tool")[0]
+        dna_plate_ids = db.extract_sample_plates(
+            [plate_id, plate_id_2], 'test', robot['name'], kit['name'],
+            tool['name'])
+        for p_id in dna_plate_ids:
+            self._clean_up_funcs.insert(
+                0, partial(db.delete_dna_plate, p_id))
+
+        # Create the target gene plates
+        plate_links = [
+            {'dna_plate_id': dna_plate_ids[0], 'primer_plate_id': 1},
+            {'dna_plate_id': dna_plate_ids[1], 'primer_plate_id': 2}]
+        exp_robot = db.get_property_options("processing_robot")[0]
```

Comment:
Those where just used in the test to try to make them independent of the contents of the DB, but given that the values are present in the patch I think I can safely use those rather than the 0. Changed.